File: models/markerSqlModel.ts
import {
  DataTypes,
  Model,
  ModelStatic
} from 'sequelize'
import sequelize from '@/config/sequelize'
import MarkerSqlRecord from '@/types/MarkerSqlRecord'
import createId from '@/utilities/createId'
const markerSqlModel: ModelStatic<Model<MarkerSqlRecord>> = sequelize.models.Marker ?? sequelize.define<Model<MarkerSqlRecord>>(
  'Marker', {
    ...createId(),
    user_id: {
      type: DataTypes.UUID,
      allowNull: true,
      references: {
        model: 'User',
        key: 'id'
      }
    },
    timeline: {
      type: DataTypes.TEXT,
      allowNull: false,
      unique: true
    },
    last_read_id: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    lock_version: {
      type: DataTypes.FLOAT,
      allowNull: false
    }
  }, {
    tableName: 'markers',
    timestamps: true
  }
)
export default markerSqlModel

File: nomad/models/marker_sql_model.py
"""
Marker SQL table model
"""
from uuid import uuid4
from django.db import models
class Marker(models.Model):
    """
    Marker SQL table model
    """
    class Meta:
        """
        Marker SQL table model meta class
        """
        verbose_name = 'Marker'
        verbose_name_plural = f'{verbose_name}s'
        db_table = verbose_name_plural.lower()
    id = models.UUIDField(
        primary_key = True,
        default = uuid4,
        editable = False
    )
    user_id = models.ForeignKey(
        to = 'auth.User',
        on_delete = models.CASCADE,
        related_name = 'markers',
        null = True,
        blank = True
    )
    timeline = models.TextField(
        unique = True
    )
    last_read_id = models.ForeignKey(
        to = 'Post',
        on_delete = models.DO_NOTHING,
        related_name = 'markers',
        null = True,
        blank = True
    )
    lock_version = models.PositiveBigIntegerField()
    created_at = models.DateTimeField(
        auto_now_add = True
    )
    updated_at = models.DateTimeField(
        auto_now = True
    )
